src/basic/boot/FStar.Util.fs

```diff
@@ -919,14 +921,19 @@ let internal hints_db_from_json_db (jdb : json_db) : hints_db =
         if  h = null then None
         else let ha = h :?> System.Object [] in
              if (Array.length ha) = 0 then None else
-                if (Array.length ha) <> 6 then failwith "malformed hint" else
-                Some {
+                if (Array.length ha) <> 6 && (Array.length ha) <> 7
+                then failwith "malformed hint"
```

Comment:
Can we raise an error instead?

Reply:
That's what the old implementation did... what kind of error would you prefer?

Reply:
I'm not sure. Either it's OK to continue without the hint, in which case we could simply issue a warning and go on; or we want to fail, but in that case it'd be nice to include the file name of the hint file.

I noticed this while reading through, but indeed I wasn't criticizing a new addition :)